graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/DgsGraphQLSourceBuilder.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.springgraphql
+
+import com.netflix.graphql.dgs.internal.DefaultDgsQueryExecutor
+import com.netflix.graphql.dgs.internal.DgsSchemaProvider
+import graphql.schema.GraphQLSchema
+import graphql.schema.TypeResolver
+import graphql.schema.idl.RuntimeWiring
+import graphql.schema.idl.TypeDefinitionRegistry
+import org.springframework.core.io.Resource
+import org.springframework.graphql.execution.*
+import org.springframework.graphql.execution.GraphQlSource.SchemaResourceBuilder
+import org.springframework.lang.Nullable
+import java.util.function.BiFunction
+import java.util.function.Consumer
+
+class DgsGraphQLSourceBuilder(private val dgsSchemaProvider: DgsSchemaProvider, private val reloadSchemaIndicator: DefaultDgsQueryExecutor.ReloadSchemaIndicator) : AbstractGraphQlSourceBuilder<SchemaResourceBuilder>(), SchemaResourceBuilder {
```

Comment:
```suggestion
class DgsGraphQLSourceBuilder(private val dgsSchemaProvider: DgsSchemaProvider, private val reloadSchemaIndicator: DefaultDgsQueryExecutor.ReloadSchemaIndicator) : AbstractGraphQlSourceBuilder<DgsGraphQLSourceBuilder>() {
```

Looks good extending `AbstractGraphQlSourceBuilder`! You can now also drop the `SchemaResourceBuilder` sub-interface with its methods related to building `GraphQLSchema`. In this implementation that's done through `DgsSchemaProvider`, and they are not relevant.

Reply:
Thanks for taking a look! Actually we realized we still need the SchemaResourceBuilder because third party libraries (multipart file uploads that you recommend as well)set up a schema resource on the builder. So we do need to inject schema resources from the builder into the DgsSchemaProvider prior to calling it.

Reply:
Right, I see now. I was also going to bring up something related to this. 

`DgsSpringGraphQLAutoConfiguration` is declared to run before `DgsAutoConfiguration` (I presume so it can declare `DgsComponent`s for the latter to detect). At the same time, the `graphQlSource` bean  needs `DgsSchemaProvider` that comes from the latter, but that's supposed to run after. There is a sort of circular relationship there, and while it doesn't seem to cause a problem currently, I think it would be good to split the config in two distinct parts.

The first part focuses on `GraphQLSchema` creation, declares `@DgsComponent`s, and is ordered before `DgsAutoConfiguration`. The second part creates the `graphQlSource` bean and is ordered after  `DgsAutoConfiguration` with the `DgsSchemaProvider`. The first config can detect `GraphQlSourceBuilderCustomizer`s such as the one from FileUpload library, and bridge any `SchemaResourceBuilder` registrations to `DgsSchemaProvider` along the lines of what `DgsRuntimeWiringConfigurerBridge` and `DgsTypeDefinitionConfigurerBridge` currently do. For the second config, the `DgsGraphQLSourceBuilder` uses `DgsSchemaProvider` and implements only the base `GraphQlSource.Builder`.